src/cmd/migrate.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-restricted-syntax */
+import { diff } from 'deep-diff'
+import { cloneDeep, get, isEqual, set, unset } from 'lodash'
+import { compare, valid } from 'semver'
+import { Argv } from 'yargs'
+import { $ } from 'zx'
+import { prepareEnvironment } from '../common/cli'
+import { OtomiDebugger, terminal } from '../common/debug'
+import { hfValues } from '../common/hf'
+import { getFilename, loadYaml, rootDir } from '../common/utils'
+import { writeValues } from '../common/values'
+import { BasicArguments, setParsedArgs } from '../common/yargs'
+import { askYesNo } from '../common/zx-enhance'
+
+const cmdName = getFilename(__filename)
+const debug: OtomiDebugger = terminal(cmdName)
+
+interface Change {
+  version: string
+  deletions?: Array<string>
+  locations?: Array<{
+    [oldLocation: string]: string
+  }>
+  mutations?: Array<{
+    [preMutation: string]: string
+  }>
+}
+
+export type Changes = Array<Change>
+
+export const moveGivenJsonPath = (yaml: Record<string, unknown> | undefined, lhs: string, rhs: string): void => {
+  const moveValue = get(yaml, lhs, 'err!')
+  if (yaml && unset(yaml, lhs)) set(yaml, rhs, moveValue)
+}
+
+export const inlineGucci = async (): Promise<string> => {
+  const execTmpl = await $`echo '{{ print "bla" }}' | gucci`
+  return execTmpl.stdout.trim()
+}
+
+// save work https://stackoverflow.com/questions/11488014/asynchronous-process-inside-a-javascript-for-loop
+
+export function filterChanges(currentVersion: string, changes: Changes): Changes {
+  if (!valid(currentVersion))
+    throw new Error(`Please set otomi.version to a valid SemVer, e.g. 1.2.3 (received ${currentVersion})`)
+  return changes.filter((c) => compare(c.version, currentVersion) >= 1).sort((a, b) => compare(a.version, b.version))
+}
+
+export const migrate = async (values: Record<string, unknown> | undefined, changes: Changes): Promise<void> => {
+  for (const c of changes) {
+    c.deletions?.forEach((del) => unset(values, del))
+    c.locations?.forEach((loc) => moveGivenJsonPath(values, Object.keys(loc)[0], Object.values(loc)[0]))
+
+    if (c.mutations && values)
```

Comment:
If existence of values is of concern here, then perhaps a check should exist on line 1 of this func